nuxeo-services/nuxeo-platform-convert/src/test/java/org/nuxeo/ecm/platform/convert/tests/TestZip2HtmlConverter.java

```diff
@@ -0,0 +1,45 @@
+package org.nuxeo.ecm.platform.convert.tests;
+
+import org.junit.Test;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.blobholder.BlobHolder;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class TestZip2HtmlConverter extends BaseConverterTest{
+
+    @Test
+    public void testZipConverter() throws Exception {
+        testConvertToMimeType("application/zip");
+        testConvertToMimeType("application/x-zip-compressed");
+    }
+
+    public void testConvertToMimeType(String mimeType) throws IOException {
+        String converterName = cs.getConverterName(mimeType, "text/html");
+        assertEquals("zip2html", converterName);
+
+        checkConverterAvailability(converterName);
+
+        BlobHolder htmlBH = getBlobFromPath("test-docs/hello.zip");
+        htmlBH.getBlob().setMimeType(mimeType);
+        Map<String, Serializable> parameters = new HashMap<>();
+
+        BlobHolder result = cs.convert(converterName, htmlBH, parameters);
+        assertNotNull(result);
+
+        List<Blob> blobs = result.getBlobs();
+        assertNotNull(blobs);
+        assertEquals(3, blobs.size());
+
+        String content = blobs.get(0).getString();
+        assertTrue(content.contains("Hello") || content.contains("hello"));
```

Comment:
Can you do a more precise check? Comparing the string with exactly what you expect? You should check a specific blob as well, not just the first one (not sure you'll have a deterministic order).

Reply:
Yes, you are correct. I do think the order is deterministic. I'll improve it.